### PR TITLE
Handle share links

### LINKED ISSUE #8: Figure out how to handle shortened share links
Reddit's gone full dumb with shareable links that look like `https://reddit.com/r/comics/s/95j2GYWoSR`.

This format seems to come only from their mobile apps, but it's damned useless for parsing in a chat bot. There's no obvious relationship between (any part of) the short "keyword" and the submission/comment ID it links to.

It's possible to handle these [by using a `Host: www.reddit.com` header](https://github.com/praw-dev/praw/issues/1968#issuecomment-1712787947) to skip an unnecessary redirect and then parse the `Location` header returned by `https://www.reddit.com/r/comics/s/95j2GYWoSR` for the actual submission/comment ID, but that's pretty obnoxious.

### PR DESCRIPTION
**Pull request by half-duplex**
Fixes #8 

Test links (from the praw commit, I'm not touching the reddit app):
https://www.reddit.com/r/redditdev/s/WNauetbiNG (submission)
https://www.reddit.com/r/redditdev/s/nGnQE1QkLC (comment)

Example output:
```
me> https://www.reddit.com/r/redditdev/s/WNauetbiNG
Sopel> [reddit] [PRAW] HTTPS enabled PRAW testing needed ('PRAW' flair) (self.redditdev) | 11 points (84.0%) | 2 comments | Posted by bboe | Created at 2014-09-17 - 06:34:31 UTC | https://redd.it/2gmzqe
me> https://www.reddit.com/r/redditdev/s/nGnQE1QkLC
Sopel> [reddit] Comment by bboe | 1 point | Posted at 2014-09-18 - 03:13:54 UTC | https://reddit.com/comments/2gmzqe/c/cklhv0f | Yes it does. That fix is also in the master branch, we just haven't made a release for it (and we probably won't until 3.0).
```

### CODE REVIEW

File: sopel_reddit/__init__.py [hunk 1]
```diff
@@ -157,18 +158,31 @@ def video_info(bot, trigger, match):
...
 
 
+@plugin.url(share_url)
+@plugin.output_prefix(PLUGIN_OUTPUT_PREFIX)
+def share_info(bot, trigger):
+    redirect_response = requests.get(
+        trigger, allow_redirects=False, headers={"User-Agent": USER_AGENT}
+    )
+    if redirect_response.status_code != 301:
+        return
```
Comment (dgw) [inline sopel_reddit/__init__.py:168]: I'd like to point out that this won't correctly handle the `http://` links allowed by the `share_url` pattern. Those get a 307 to their `https://` counterpart (still with the `/s/tracking_gibberish` part unresolved).

Probably not too bad to move some of this logic into a separate function that can be called recursively. (I remember you said earlier today that reddit blocked your IP during testing. Was this check originally added because of that?)

Comment (half-duplex) [reply sopel_reddit/__init__.py:168]: The http->https redirects are 301s for me, but yeah, unhandled case.

The status_code check was mostly for "if this isn't a redirect, the rest of this definitely won't work". The IP ban is a 403.

Comment (half-duplex) [reply sopel_reddit/__init__.py]: All urls shall become https :grin:

**Comment by dgw** [comment]
I mentioned this on IRC already, but it belongs in the PR:

Reddit sometimes blocks the request this performs with an HTTP 403 Forbidden status. The code works for me locally on a residential connection, and I presume @half-duplex is also on one. Testing in "production" on our main bot (running on a DigitalOcean server) did NOT work, however. As currently written it will simply fail silently.

`praw` finally released support for these stupid URLs in October with version 7.8.0. It'd require reworking the plugin logic a bit, but using their method (passing the share link in `Reddit.comment(url='link')` or `Reddit.submission(url='link')` works on the server, where this PR's current code receives a 403.

Do you feel like experimenting with praw 7.8.x and its built-in share link resolution feature?

**Comment by dgw** [comment]
It's easier to show my pseudocode idea here than on IRC, because line breaks:

```python
url = trigger.match.group(0)
comment = submission = None

try:
    comment = bot.memory['reddit_praw'].comment(url=url)
except praw.exceptions.InvalidURL:
    submission = bot.memory['reddit_praw'].submission(url=url)

if comment:
    show_comment_info()  # needs a little reworking to optionally take the comment object instead of an ID
else:
    show_post_info()  # ditto for submission object instead of ID
```

Do I love this? Absolutely not. I just think it's less fragile to use PRAW's built-in handling, because that seems to work on the server where directly resolving the redirect gets rejected with a 403.

Yes, I wish PRAW had a saner API for this. The whole reason we hate the `/s/` links is because they obfuscate what type of item is being linked to, so requiring the client to know which type it's expecting is silly.

**Comment by half-duplex** [comment]
Two commits so the diff is readable, the second is just a big unindent.

```
me> https://www.reddit.com/r/redditdev/s/WNauetbiNG submission
Sopel> [reddit] [PRAW] HTTPS enabled PRAW testing needed ('PRAW' flair) (https://redd.it/2gmzqe) to r/redditdev | 9 points (82.0%) | 2 comments | Posted by bboe | Created at 2014-09-17 - 06:34:31 UTC
me> https://www.reddit.com/r/redditdev/s/nGnQE1QkLC comment
Sopel> [reddit] Comment by bboe | 1 point | Posted at 2014-09-18 - 03:13:54 UTC | https://reddit.com/comments/2gmzqe/c/cklhv0f | Yes it does. That fix is also in the master branch, we just haven't made a release for it (and we probably won't until 3.0).
me> https://www.reddit.com/r/redditdev/s/nGn broken
Sopel> me: Error fetching metadata
```

**Comment by dgw** [comment]
Is this even worth rebasing any more? See [draconian announcement](https://www.reddit.com/r/redditdev/comments/1oug31u/introducing_the_responsible_builder_policy_new/) regarding API access made today.

tl;dr ALL new app IDs will require manual approval by reddit admins, which makes the future-proofing from #1 moot for anyone who doesn't already have one.

File: sopel_reddit/plugin.py [hunk 2]
```diff
@@ -178,97 +204,120 @@ def rgallery_info(bot, trigger, match):
     return say_post_info(bot, trigger, match.group(1), False)
 
 
-def say_post_info(bot, trigger, id_, show_link=True, show_comments_link=False):
+def say_post_info(
+    bot: SopelWrapper,
+    trigger: Trigger,
+    id_: Optional[str] = None,
+    url: Optional[str] = None,
```
Comment (dgw) [inline sopel_reddit/plugin.py]: Upstream we have moved to `str | None` style. Since this plugin requires 8.0+ (on `master`, at least; I didn't check PyPI) it should be safe to do that here as well.

```suggestion
    id_: str | None = None,
    url: str | None = None,
```

File: sopel_reddit/plugin.py [hunk 3]
```diff
@@ -178,97 +204,120 @@ def rgallery_info(bot, trigger, match):
...
+        comments_link=comments_link)
+
+    bot.say(message)
+
+
+def say_comment_info(
+    bot: SopelWrapper,
+    trigger: Trigger,
+    id_: Optional[str] = None,
+    url: Optional[str] = None,
```
Comment (dgw) [inline sopel_reddit/plugin.py]: Same as the other spot in `say_post_info()`.

```suggestion
    id_: str | None = None,
    url: str | None = None,
```

File: sopel_reddit/plugin.py [hunk 4]
```diff
@@ -23,6 +24,12 @@
 from sopel.tools import time
 from sopel.tools.web import USER_AGENT
 
+if TYPE_CHECKING:
+    from typing import Optional
+
```
Comment (dgw) [inline sopel_reddit/plugin.py]: Not needed if the two suggested changes below regarding use of `|` are accepted.

```suggestion
```

File: sopel_reddit/plugin.py [hunk 5]
```diff
@@ -157,6 +165,24 @@ def video_info(bot, trigger, match):
...
+        say_comment_info(bot, trigger, url=url, show_link=True)
+        return
+    except praw.exceptions.InvalidURL:
+        pass
+    except AssertionError:
+        # Invalid share links give "AssertionError: Unexpected status code: 307"
+        bot.reply("Error fetching metadata")
+        return
+
+    say_post_info(bot, trigger, url=url)
```
Comment (dgw) [inline sopel_reddit/plugin.py]: Part of handling the share link should be to give an alternative clickable URL without the tracking built in. The earlier call to `say_comment_info()` does that, but this wouldn't without:

```suggestion
    say_post_info(bot, trigger, url=url, show_comments_link=True)
```

File: sopel_reddit/plugin.py [hunk 6]
```diff
@@ -178,97 +204,120 @@ def rgallery_info(bot, trigger, match):
...
+    percent = color('{:.1%}'.format(s.upvote_ratio), point_color)
+
+    comments_text = 'comment' if s.num_comments == 1 else 'comments'
+
+    comments_link = ''
+    if show_comments_link:
+        try:
+            comments_link = ' | ' + s.shortlink
+        except AttributeError:
+            # the value assigned earlier will be used
```
Comment (dgw) [inline sopel_reddit/plugin.py]: I think this comment is actually more confusing than not having it. (No, you didn't invent the comment; it just sent me on a wild goose chase thinking there was another assignment to `comments_link` earlier, before `if show_comments_link:`.)

```suggestion
```

File: sopel_reddit/plugin.py [hunk 7]
```diff
@@ -157,6 +165,24 @@ def video_info(bot, trigger, match):
...
+
+    try:
+        say_comment_info(bot, trigger, url=url, show_link=True)
+        return
+    except praw.exceptions.InvalidURL:
+        pass
+    except AssertionError:
+        # Invalid share links give "AssertionError: Unexpected status code: 307"
+        bot.reply("Error fetching metadata")
+        return
```
Comment (dgw) [inline sopel_reddit/plugin.py:179]: Would much prefer a specific error type, but if all `praw` gives us is the `AssertionError`, then we're stuck with it. I couldn't find a relevant `assert` line in PRAW code, though, which is odd. (Could also be me not understanding the layout/class hierarchy.)

I did a little testing and discovered that a share link can become "invalid" (trip this `except` block) if you modify the `subreddit` part after `/r/`, which is asinine but—again—out of our control. Maybe reddit combines the sub name + share ID to avoid collisions, idk.

File: sopel_reddit/plugin.py [hunk 8]
```diff
@@ -178,97 +204,120 @@ def rgallery_info(bot, trigger, match):
...
-                bot.kick(
-                    trigger.nick, trigger.sender,
-                    'Linking to spoiler content in a spoiler-free channel.'
-                )
-
-        if s.author:
-            author = s.author.name
-        else:
-            author = '[deleted]'
+            s = bot.memory["reddit_praw"].submission(id=id_)
```
Comment (dgw) [inline sopel_reddit/plugin.py]: I don't love that nested-if structure.

```py
        if id_:
            s = bot.memory["reddit_praw"].submission(id=id_)
        elif url:
            s = bot.memory["reddit_praw"].submission(url=url)
        else:
            raise TypeError("Expected either id_ or url parameter")
```

I'm looking at this as a single block of added lines (in Split mode), but internally the diff has a bunch of deletions in between so I can't make  one-click suggestion to apply.

File: sopel_reddit/plugin.py [hunk 9]
```diff
@@ -178,97 +204,120 @@ def rgallery_info(bot, trigger, match):
...
+    show_link: bool = False,
+):
     try:
-        c = bot.memory['reddit_praw'].comment(id_)
+        if id_ is None:
+            if url is None:
+                raise TypeError("Expected either id_ or url parameter")
+            c = bot.memory["reddit_praw"].comment(url=url)
+        else:
+            c = bot.memory["reddit_praw"].comment(id=id_)
```
Comment (dgw) [inline sopel_reddit/plugin.py]: Same tune as in `say_post_info()` re: the shape of if/else.

```py
        if id_:
            c = bot.memory["reddit_praw"].comment(id=id_)
        elif url:
            c = bot.memory["reddit_praw"].comment(url=url)
        else:
            raise TypeError("Expected either id_ or url parameter")
```

File: sopel_reddit/plugin.py [hunk 10]
```diff
@@ -279,12 +328,20 @@ def say_comment_info(bot, trigger, id_):
 
     posted = get_time_created(bot, trigger, c.created_utc)
 
+    # DIY shortish-link, since c.permalink is both long and not a link
```
Comment (dgw) [inline sopel_reddit/plugin.py]:
```suggestion
    # DIY short-ish link, since c.permalink is both long and not a link
```

File: sopel_reddit/plugin.py [hunk 11]
```diff
@@ -279,12 +328,20 @@ def say_comment_info(bot, trigger, id_):
...
+    link = ""
+    if show_link:
+        link = "https://reddit.com/comments/{}/c/{} | ".format(
+            c.link_id[3:],  # strip t3_ prefix
+            c.id
```
Comment (dgw) [inline sopel_reddit/plugin.py]: Style nit

```suggestion
            c.id,
```

**Review by dgw — CHANGES_REQUESTED** [review]
I admit, I switched on "Hide whitespace" halfway through this to make the diff less confusing.

File: sopel_reddit/plugin.py (hunk 5 above)
Comment (half-duplex) [reply sopel_reddit/plugin.py]: In one or another revision I guess I fixed this in a different way but didn't update the top post. Current:
```
<mal> https://www.reddit.com/r/redditdev/s/WNauetbiNG
<sopel> [reddit] [PRAW] HTTPS enabled PRAW testing needed ('PRAW' flair) (https://redd.it/2gmzqe) to r/redditdev | 11 points (84.0%) | 2 comments | Posted by bboe | Created at 2014-09-17 - 06:34:31 UTC
<mal> https://www.reddit.com/r/redditdev/s/nGnQE1QkLC
<sopel> [reddit] Comment by bboe | 1 point | Posted at 2014-09-18 - 03:13:54 UTC | https://reddit.com/comments/2gmzqe/c/cklhv0f | Yes it does. That fix is also in the master branch, we just haven't made a release for it (and we probably won't until 3.0).
```
Adding `show_comments_link=True` to this revision causes a *second* link to be added at the end of responses to post-type share links.

File: sopel_reddit/plugin.py (hunk 7 above)
Comment (half-duplex) [reply sopel_reddit/plugin.py:179]: Traceback tail:
```
  File ".venv/lib/python3.13/site-packages/prawcore/sessions.py", line 328, in request
    return self._request_with_retries(...
  File ".venv/lib/python3.13/site-packages/prawcore/sessions.py", line 271, in _request_with_retries
    response.status_code in self.SUCCESS_STATUSES
AssertionError: Unexpected status code: 307
```

File: sopel_reddit/plugin.py (hunk 8 above)
Comment (half-duplex) [reply sopel_reddit/plugin.py]: Good call, it can go even lighter I think:
```py
if not (id_ or url):
  raise TypeError("Expected either id_ or url parameter")
c = bot.memory["reddit_praw"].submission(id=id_, url=url)
```

**Comment by half-duplex** [comment]
Also moved one of the new `url` arguments to the end since I realized there were several callsites with ordered arguments for the other optional args.

File: sopel_reddit/plugin.py (hunk 7 above)
Comment (dgw) [reply sopel_reddit/plugin.py:179]: Aha, I pointed GH code search at `praw` instead of `prawcore`.

**Comment by dgw** [comment]
> Also moved one of the new `url` arguments to the end since I realized there were several callsites with ordered arguments for the other optional args.

Feel free to just update the callsites and order function parameters in the way that makes the most sense. This plugin's callables aren't a public API :)

File: sopel_reddit/plugin.py (hunk 5 above)
Comment (dgw) [reply sopel_reddit/plugin.py]: I… am going to take another spin through the code and hopefully figure that out. Can't do it immediately, so I just marked this thread as unresolved.

**Comment by half-duplex** [comment]
Eh, fair, I'll make the def arg list prettier (and put the related id/url args together)

File: sopel_reddit/plugin.py (hunk 5 above)
Comment (half-duplex) [reply sopel_reddit/plugin.py]: Say_post_info's show_link defaults to True.

I noticed I might have broken the "self.subredditname" thing, but that seems to only exist on old.reddit, so I'm just going to remove it.

This also made me double check show_link vs show_comments_link, which the s.url->s.shortlink change made redundant. Callsites setting either of them are:
- image_info - (False if not preview.redd.it) / True
- video_info - False/True (unchanged)
- rgallery_info - False/False (so doesn't matter)

For preview.redd.it links, s.url is the matching i.redd.it link, so I'm going to rename and shuffle that logic a bit... And make it not add shortlinks for posted shortlinks, and make it add commas to big numbers. Will try to split these to another PR if you want.

Original:
>mal> `https://i.redd.it/05cnhtc5yo331.jpg`
sopel> [reddit] keanu reeves ascends to save mankind to r/AccidentalRenaissance | 61586 points (93.0%) | 539 comments | Posted by worstwerewolf | Created at 2019-06-11 - 08:50:58 UTC | `https://redd.it/bza0vr`
mal> `https://preview.redd.it/05cnhtc5yo331.jpg`
sopel> [reddit] keanu reeves ascends to save mankind (`https://i.redd.it/05cnhtc5yo331.jpg`) to r/AccidentalRenaissance | 61585 points (93.0%) | 539 comments | Posted by worstwerewolf | Created at 2019-06-11 - 08:50:58 UTC | `https://redd.it/bza0vr`

Now:
>mal> `https://i.redd.it/05cnhtc5yo331.jpg`
sopel> [reddit] keanu reeves ascends to save mankind to r/AccidentalRenaissance | 61,597 points (93.0%) | 539 comments| Post by worstwerewolf at 2019-06-11 - 08:50:58 UTC | `https://redd.it/bza0vr`
mal> `https://preview.redd.it/05cnhtc5yo331.jpg`
sopel> [reddit] keanu reeves ascends to save mankind to r/AccidentalRenaissance | 61,594 points (93.0%) | 539 comments| Post by worstwerewolf at 2019-06-11 - 08:50:58 UTC | `https://redd.it/bza0vr` | Full: `https://i.redd.it/05cnhtc5yo331.jpg`
mal> `https://www.reddit.com/r/redditdev/s/WNauetbiNG`
sopel> [reddit] [PRAW] HTTPS enabled PRAW testing needed ('PRAW' flair) to r/redditdev | 13 points (94.0%) | 2 comments| Post by bboe at 2014-09-17 - 06:34:31 UTC | `https://redd.it/2gmzqe`
mal> `https://v.redd.it/xcre0omf86j21`
sopel> [reddit] Not Again ('MEDIA' flair) to r/FortNiteBR | 11,786 points (92.0%) | 539 comments| Post by Spike36O at 2019-02-27 - 20:34:25 UTC | `https://redd.it/avi22e`
mal> `https://redd.it/2gmzqe`
sopel> [reddit] [PRAW] HTTPS enabled PRAW testing needed ('PRAW' flair) to r/redditdev | 10 points (87.0%) | 2 comments| Post by bboe at 2014-09-17 - 06:34:31 UTC

Comment (dgw) [reply sopel_reddit/plugin.py]: > I noticed I might have broken the "self.subredditname" thing, but that seems to only exist on old.reddit, so I'm just going to remove it.

tbh I would like this plugin not to acknowledge that "new reddit" exists, so bonus points if we can keep the `self.subredditname` thing going. The only reddit is old reddit! 🤪

Did you mean to remove the submission link (as in the URL OP submitted) from regular threads like [this](https://www.reddit.com/r/pcmasterrace/comments/1tvubc6/nand_flash_makers_earned_a_record_46_billion_in/)? I don't mind some of the format tweaks—it looks good—but that's certainly losing a piece of information that we're used to seeing from the plugin.

```
<v1.1.0> [reddit] NAND flash makers earned a record $46 billion in revenues over the first
         quarter of 2026, a shocking 3.5 times more than last year ('News/Article' flair)
         (https://www.pcgamer.com/hardware/ssds/nand-flash-makers-earned-a-record-usd46-
         billion-in-revenues-over-the-first-quarter-of-2026-a-shocking-3-5-times-more-than
         -last-year/) to r/pcmasterrace | 1317 points (98.0%) | 90 comments | Posted by
         HatingGeoffry | Created at 2026-06-03 - 16:19:25 UTC
<f88c916> [reddit] NAND flash makers earned a record $46 billion in revenues over the first
          quarter of 2026, a shocking 3.5 times more than last year ('News/Article' flair)
          to r/pcmasterrace | 1,319 points (98.0%) | 90 comments| Post by HatingGeoffry at
          2026-06-03 - 16:19:25 UTC | https://redd.it/1tvubc6
```

We should also look at truncation behavior, but that's definitely "new PR" territory—for the next maintenance release (1.2.1, not 1.2.0).

Comment (half-duplex) [reply sopel_reddit/plugin.py]: I never really understood the "self post" thing in the first place, and it meant even less after self-posts allowed images, but fair enough.
I intentionally switched from the long link to the short link since the long reddit links just waste characters duplicating the post title, but that was because apparently all of my test links were self-posts... I'll re-add non-reddit long links (all non-self posts, I guess?).
Any thoughts on whether long links should go at the end to make the rest of the info more readable, or where they were because the points/comments/date are less interesting/important?

Comment (dgw) [reply sopel_reddit/plugin.py]: Yes, the links for non-self posts (and I guess non-media posts, too? except you might have a case for putting the `/gallery` link right there idfk, reddit has too many post types). It can be the last thing on the line, that's OK.

I don't think the previous behavior was to include a long link to the post itself (`reddit.com/r/subredditname/comments/post_title_slug`); it already used `s.shortlink` for that, when `show_comments_link` is True.

https://github.com/sopel-irc/sopel-reddit/blob/48cf7f5ae7805981ee3de212377d6a7d23f3f3d4/sopel_reddit/plugin.py#L243-L245

`show_link` on the other hand was always about showing the article URL, or the `v.redd.it` or `i.redd.it` link _independently of_ whether the comments link (`redd.it/postID`) needed showing.

Comment (half-duplex) [reply sopel_reddit/plugin.py]: Okay, let's try this one (114b7546f3f48ee7838f32bf353e13c4b4595e61)

I don't really see a benefit to showing the shortlink for non-tracking links, you can just click the link the user posted. For share links and non-comments-page links it's shown now.

Full URLs are back:
> mal> `https://www.reddit.com/r/pcmasterrace/comments/1tvubc6/nand_flash_makers_earned_a_record_46_billion_in/`
sopel> [reddit] NAND flash makers earned a record $46 billion in revenues over the first quarter of 2026, a shocking 3.5 times more than last year [News/Article] to r/pcmasterrace | 1,997 points (98.0%) | 109 comments | Post by HatingGeoffry at 2026-06-03 - 16:19:25 UTC | `https://www.pcgamer.com/hardware/ssds/nand-flash-makers-earned-a-record-usd46-billion-in-revenues-over-the-first-quarter-of-2026-a-shocking-3-5-times-more-than-last-year/`

This *would* have ` | https://redd.it/1tvubc6` on the end if not for truncation:
> mal> `https://www.reddit.com/r/pcmasterrace/s/dThznIuHuh`
sopel> [reddit] NAND flash makers earned a record $46 billion in revenues over the first quarter of 2026, a shocking 3.5 times more than last year [News/Article] to r/pcmasterrace | 2,004 points (98.0%) | 109 comments | Post by HatingGeoffry at 2026-06-03 - 16:19:25 UTC | `https://www.pcgamer.com/hardware/ssds/nand-flash-makers-earned-a-record-usd46-billion-in-revenues-over-the-first-quarter-of-2026-a-shocking-3-5-times-more-than-last-year/`

Self-posts are self.whatever:
> mal> `https://www.reddit.com/r/redditdev/s/WNauetbiNG`
sopel> [reddit] [PRAW] HTTPS enabled PRAW testing needed [PRAW] to self.redditdev | 12 points (94.0%) | 2 comments | Post by bboe at 2014-09-17 - 06:34:31 UTC | `https://redd.it/2gmzqe`

Image posts behave:
> mal> `https://www.reddit.com/r/pcmasterrace/s/HBLbiZlLW3`
sopel> [reddit] Best sleep ever [Meme/Macro] to r/pcmasterrace | 5,942 points (86.0%) | 520 comments | Post by Sus_Tomato at 2026-06-03 - 19:21:52 UTC | `https://i.redd.it/j9fpsncsb45h1.jpeg` | `https://redd.it/1tvzrkz`
mal> `https://i.redd.it/j9fpsncsb45h1.jpeg`
sopel> [reddit] Best sleep ever [Meme/Macro] to r/pcmasterrace | 5,942 points (86.0%) | 520 comments | Post by Sus_Tomato at 2026-06-03 - 19:21:52 UTC | `https://redd.it/1tvzrkz`
mal> `https://redd.it/1tvzrkz`
sopel> [reddit] Best sleep ever [Meme/Macro] to r/pcmasterrace | 5,946 points (86.0%) | 520 comments | Post by Sus_Tomato at 2026-06-03 - 19:21:52 UTC | `https://i.redd.it/j9fpsncsb45h1.jpeg`

Video posts behave:
> mal> `https://www.reddit.com/r/FortNiteBR/comments/avi22e/not_again/`
sopel> [reddit] Not Again [MEDIA] to r/FortNiteBR | 11,787 points (92.0%) | 539 comments | Post by Spike36O at 2019-02-27 - 20:34:25 UTC | `https://v.redd.it/xcre0omf86j21`
mal> `https://redd.it/avi22e`
sopel> [reddit] Not Again [MEDIA] to r/FortNiteBR | 11,788 points (92.0%) | 539 comments | Post by Spike36O at 2019-02-27 - 20:34:25 UTC | `https://v.redd.it/xcre0omf86j21`
mal> `https://v.redd.it/xcre0omf86j21`
sopel> [reddit] Not Again [MEDIA] to r/FortNiteBR | 11,783 points (92.0%) | 539 comments | Post by Spike36O at 2019-02-27 - 20:34:25 UTC | `https://redd.it/avi22e`

Galleries:
> mal> `https://www.reddit.com/r/drawing/comments/1tw50ye/brush_pen_study/`
sopel> [reddit] Brush pen study [ink] to r/drawing | 4 points (100.0%) | 1 comment | Post by ColombianNYC at 2026-06-03 - 22:33:58 UTC | `https://www.reddit.com/gallery/1tw50ye`
mal> `https://www.reddit.com/gallery/1tw50ye`
sopel> [reddit] Brush pen study [ink] to r/drawing | 4 points (100.0%) | 1 comment | Post by ColombianNYC at 2026-06-03 - 22:33:58 UTC

Though, actually clicking some of these links now... is there any case where the video or gallery link brings you somewhere other than the comments page?

Comment (dgw) [reply sopel_reddit/plugin.py]: > Though, actually clicking some of these links now... is there any case where the video or gallery link brings you somewhere other than the comments page?

Maybe not, but this PR has been in review hell for long enough! If we see more changes to make, including tweaks to when the full URL gets posted for videos/galleries, that can be a separate patch. There are probably little tweaks to make re: truncation once we observe more of the plugin's new behavior "in the wild" with a wide sampling of different posts.

**Review by dgw — APPROVED** [review]
Huh, I suppose I should actually "approve" this so my previous request-changes marker goes away.
